src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,203 @@
+    ngOnDestroy() {
+        document.removeEventListener('keydown', this.handleKeyboardEvents);
+        window.removeEventListener('resize', this.resizeCanvasBasedOnContainer);
+    }
+
+    private async loadPdf(fileUrl: string) {
+        try {
+            const loadingTask = PDFJS.getDocument(fileUrl);
+            const pdf = await loadingTask.promise;
+            this.totalPages = pdf.numPages;
+
+            for (let i = 1; i <= pdf.numPages; i++) {
+                const page = await pdf.getPage(i);
+                const viewport = page.getViewport({ scale: 1 });
+                const canvas = this.createCanvas(viewport);
+                const context = canvas.getContext('2d');
+                if (context) {
+                    await page.render({ canvasContext: context, viewport }).promise;
+                }
+
+                const container = this.createContainer(canvas, i);
+                this.pdfContainer.nativeElement.appendChild(container);
+            }
+
+            URL.revokeObjectURL(fileUrl);
+        } catch (error) {
+            onError(this.alertService, error);
+        }
+    }
+
+    private createCanvas(viewport: PDFJS.PageViewport): HTMLCanvasElement {
+        const canvas = document.createElement('canvas');
+        canvas.height = viewport.height;
+        canvas.width = viewport.width;
+        const fixedWidth = 250;
+        const scaleFactor = fixedWidth / viewport.width;
+        canvas.style.width = `${fixedWidth}px`;
+        canvas.style.height = `${viewport.height * scaleFactor}px`;
+        return canvas;
+    }
+
+    private createContainer(canvas: HTMLCanvasElement, pageIndex: number): HTMLDivElement {
+        const container = document.createElement('div');
+        container.classList.add('pdf-page-container');
+        container.style.cssText = `position: relative; display: inline-block; width: ${canvas.style.width}; height: ${canvas.style.height}; margin: 20px; box-shadow: 0 2px 6px var(--pdf-preview-canvas-shadow);`;
+
+        const overlay = this.createOverlay(pageIndex);
+        container.appendChild(canvas);
+        container.appendChild(overlay);
+
+        container.addEventListener('mouseenter', () => (overlay.style.opacity = '1'));
+        container.addEventListener('mouseleave', () => (overlay.style.opacity = '0'));
+        overlay.addEventListener('click', () => this.displayEnlargedCanvas(canvas, pageIndex));
+
+        return container;
+    }
+
+    private createOverlay(pageIndex: number): HTMLDivElement {
+        const overlay = document.createElement('div');
+        overlay.classList.add('pdf-page-overlay');
+        overlay.innerHTML = `<span>${pageIndex}</span>`;
+        overlay.style.cssText = `position: absolute; top: 0; left: 0; width: 100%; height: 100%; display: flex; justify-content: center; align-items: center; font-size: 24px; color: white; background-color: rgba(0, 0, 0, 0.4); z-index: 1; transition: opacity 0.3s ease; opacity: 0; cursor: pointer;`;
```

Comment:
Why do we use inline css here? cant we move this to a SCSS file and just use the respective class on the element?

Reply:
The problem is when you use canvas, you need the specify the css instances before the initialization because what is drawn in the canvas is not really an HTML Element. I couldn't really find a solution for this, normal css changes are not reflected, do you know something?